appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1518,7 +1516,7 @@ private void deployAll() throws GlassFishException {
                     File file = new File(entry);
                     String deployContext = file.getName();
                     String name = deployContext.substring(0, deployContext.length() - 4);
-                    if (deployContext.endsWith(".ear") || deployContext.endsWith(".war") || deployContext.endsWith(".jar") || deployContext.endsWith(".rar")) {
+                    if (hasJavaFileExtension(deployContext)) {
```

Comment:
Shouldn't this rather be called enterprise Java file extension?

Reply:
Or hasJavaArchiveExtension.